frontend/src/App.tsx

```diff
@@ -21,6 +25,14 @@ function App() {
           <Route path="/google" element={<GoogleAuthLogin />} />
           <Route path="/bus-schedule" element={<BusSchedule />} />
           <Route path="/bus-details" element={<BusDetails />} />
+          <Route
+            path="/demo-Page"
+            element={
+              <UserProtectedRoute isAuth={isAuth}>
```

Comment:
Don't pass `isAuth` as prop. Write this logic inside the `UserProtectedRoute` component to make it independent.

Reply:
Done